tests/runtime/regression

```diff
@@ -24,8 +24,8 @@ EXPECT 0
 TIMEOUT 5
 
 NAME c_array_indexing
-RUN {{BPFTRACE}} -e 'struct Foo { int a; uint8_t b[10]; } uprobe:testprogs/uprobe_test:function2 { $foo = (struct Foo *)arg0; printf("%c %c %c %c %c\n", $foo->b[0], $foo->b[1], $foo->b[2], $foo->b[3], $foo->b[4]) }' -c ./testprogs/uprobe_test
```

Comment:
why is this test changed instead of adding a new one? This is a regression test added for a specific case so removing it seems weird

Reply:
Because the change is at language level, this is what I meant in the side effect section.

> This pr may cause some of the existing char array becoming strings, as the changes in the test files shown.
There are other cases may fail, e.g., c_array_indexing https://github.com/iovisor/bpftrace/blob/master/tests/runtime/regression#L26

In this testcase, the `b` field in `Foo` is now parsed to string type, the original test won't pass, I use the `c` field to replace, it will verify the c array indexing functionality as well.